Jenkinsfile

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env groovy
+
+pipeline {
+  agent none
+  environment {
+    BASE_DIR="src/github.com/elastic/beats"
+  }
+  options {
+    timeout(time: 1, unit: 'HOURS') 
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  parameters {
+    booleanParam(name: 'Run_As_Master_Branch', defaultValue: false, description: 'Allow to run any steps on a PR, some steps normally only run on master branch.')
+  }
+  stages {
+    /**
+     Checkout the code and stash it, to use it on other stages.
+    */
+    stage('Checkout') {
+      agent { label 'linux && immutable' }
+      environment {
+        PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+        HOME = "${env.WORKSPACE}"
+        GOPATH = "${env.WORKSPACE}"
+      }
+      options { skipDefaultCheckout() }
+      steps {
+        gitCheckout(basedir: "${BASE_DIR}")
+        stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+        script {
+          env.GO_VERSION = readFile("${BASE_DIR}/.go-version")
+        }
+      }
+    }
+    /**
+    Updating generated files for Beat.
+    Checks the GO environment.
+    Checks the Python environment.
+    Checks YAML files are generated. 
+    Validate that all updates were committed.
+    */
+    stage('Intake') { 
+      agent { label 'linux && immutable' }
+      options { skipDefaultCheckout() }
+      environment {
+        PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+        HOME = "${env.WORKSPACE}"
+        GOPATH = "${env.WORKSPACE}"
+      }
+      when { 
+        beforeAgent true
+        expression { return params.intake_ci }
+      }
+      steps {
+        withEnvWrapper() {
+          unstash 'source'
+          dir("${BASE_DIR}"){
+            sh './dev-tools/jenkins_intake.sh'
+          }
+        }
+      }
+    }
+  }
+  post {
+    success {
+      echoColor(text: '[SUCCESS]', colorfg: 'green', colorbg: 'default')
+    }
+    aborted {
+      echoColor(text: '[ABORTED]', colorfg: 'magenta', colorbg: 'default')
+    }
+    failure { 
+      echoColor(text: '[FAILURE]', colorfg: 'red', colorbg: 'default')
+      //step([$class: 'Mailer', notifyEveryUnstableBuild: true, recipients: "${NOTIFY_TO}", sendToIndividuals: false])
```

Comment:
We would probably still rely on runbld?

Reply:
Something we should look into. To start playing around with it I think not a requirement.